packages/mobile/src/exchange/WithdrawCeloQrScannerScreen.tsx

```diff
@@ -31,13 +36,20 @@ function fetchAddressFromQrData(data: string): string | null {
 function WithdrawCeloQrScannerScreen({ route }: Props) {
   const { onAddressScanned } = route.params
 
+  const { t } = useTranslation(Namespaces.global)
+  const dispatch = useDispatch()
+
   const onBarCodeDetected = memoize(
     (qrCode: QrCode) => {
       const qrData = qrCode.data
       Logger.debug(TAG, 'Bar code detected: ' + qrData)
       const address = fetchAddressFromQrData(qrData) || qrData
-      onAddressScanned(address)
-      navigateBack()
+      if (address.startsWith('0x') && address.length === 42) {
```

Comment:
Nit: defining 42 as address length constant to make it clear where it comes from